### DISCUSSION TITLE
My experience ordering and building

### DISCUSSION THREAD

**Discussion by dlavey** [rbaron/b-parasite#105] [Show and tell]
I finally got off my rear and ordered some b parasites via JLCPCB. The instructions from the wiki were pretty much spot on. I did get an email asking if the orientation on Q4 was correct, they suggested I rotate it, but I asked them to leave it as it was. I didn't want to deal with the hassle of trying to solder my own nRF52's. On a good day, I'm lucky to get a decent solder to a normal sized pad. I ended up using a [E73-2G4M08S1EX](https://jlcpcb.com/partdetail/3160500-E732G4M08S1EX/C2903762) that JLCPCB had in stock. I didn't realize until after the fact that it required an external antenna so I picked up some of [these](https://www.amazon.com/gp/product/B07JFZDYVS/) from Amazon. Even with the cost of the antenna, the price comes out cheaper than the regular specified part. I'm not sure what the practical different between the nRF52840 and the nRF52833 is, so I have no idea if the cost savings is worth it or not, but the ones I used were in stock and I didn't have to solder them myself and leave a big ole mess. Another question/issue I got from JLCPCB after they reviewed the order was that the pads on the E73-2G4M08S1EX don't match what is on the PCB. The chip has an inner ring of pads that aren't used by the b parasite and hence the pads aren't on the PCB. I told them the inner pads weren't used in the project and to just solder as is. They were fine with that. 

The setup for flashing is kind of a pain in the butt, it's not hard, there are just a lot of steps to go through. It's only a one time process though. I didn't use a J-Link programmer like is recommended, but instead used [this one](https://www.amazon.com/gp/product/B07V2CKG19/) I got from Amazon for $10. Out of the box, it won't flash the b parasite via VS Code, but I'm guessing with some config changes it would. I ended up using VS Code to build the firmware, and then manually used nrfjprog to flash the firmware. When I first tried to flash the firmware, I was getting errors from nrfjprog about access protection being enabled. To fix that, before flashing, I had to run `nrfjprog --family NRF52 --recover` and then flash with the built firmware. 

To solve the problem of holding the programmer to the pads for flashing, I designed and 3D printed a clip similar to what's linked on the wiki and it's worked quite well for me. [These](https://www.amazon.com/gp/product/B09NKCJ9WY/) are the pins I used and designed the cutout in the clip for. I soldered the wires from the programmer to the pins and then put it in the slot in the clip and smothered it in hot glue. So far it's held up fine. I did another revision on the clip to make a cutout so the clip can still be used if the battery holder has already been installed. The one shown all wired up doesn't have that cutout, but I can still connect it to the board, but it's a tight fit with the battery holder, so I notched a bit out and the test fits on the newest version seem to fit just fine with the battery holder attached. 

So far, I've flashed two of the boards and have one in use in a plant and the other I'm going to try some UV solder mask ink on the edges to see if/how well it protects. I have a b parasite and a Mi Flora in the same plant right next to each other right now and the readings are quite different as far as moisture goes. The b parasite shows 69% and the Mi Flora shows 32%, and they're right next to each other. I'm going to let things settle for a bit and and see what moisture percentage is "dry" and means I need to water the plant. I'm integrating directly with HA and the setup was easy. I still have some configuration and dashboarding to do there though. I have seen the issue with the sensors showing as "unavailable" at times, right now, this looks like a known issue and hasn't caused me too much grief, so I'm not going to worry too much about it. 

I ordered 15 pieces and the price, with UPS shipping, tax, etc., came out to $10.21 each. I ordered them on 1/19/2023 and received them on 2/6/2023. I probably would have gotten them sooner, but I ordered during the Lunar New Year. I wanted black PCB's and they weren't manufacturing parts with black PCBs during the holiday, only regular green ones. 

I've attached some pics as well as the files for the programmer clip I created, hopefully someone finds it useful. 
[3D Clip CAD and STL](https://github.com/rbaron/b-parasite/files/10690618/Clip.zip)  

**Update 1:** I forgot to mention in my original post, I had issues with VS Code and using the custom board. I followed [this](https://devzone.nordicsemi.com/f/nordic-q-a/80318/using-vscode-with-my-own-projects-and-boards/332744#332744) and was able to get the board to show up in the build settings. If you opt to use an external antenna as I did, make sure you put it on before attaching the battery cover, putting the antenna on is hard enough since they're so small, being able to easily keep the board as flat as possible makes it easier. I only have a 5x magnifier, if I had a higher power one, I'd use it while attaching the antenna's. Once the antenna is on there, put a dab of hot glue on it, otherwise you're bound to knock it off and have to re-attach it which isn't fun. 

![684A7F2E-C1FE-4094-8DB6-FECCE6D5D20D_1_105_c](https://user-images.githubusercontent.com/36577992/217674168-c5e07965-b793-46e6-a9f4-b1fb1f565fb4.jpeg)
![0806757A-5479-4F0A-AC63-3B23A00FF35A_1_105_c](https://user-images.githubusercontent.com/36577992/217674170-7cefc5d2-1b45-468b-961f-811943f0e755.jpeg)
![60875AA8-5C73-46CB-8A28-D51BD1BF2CF2](https://user-images.githubusercontent.com/36577992/217674171-9d354540-3718-45f1-9004-2b23802ca57b.jpeg)
![ACF20758-4693-49BE-8348-8D8DB323EE38](https://user-images.githubusercontent.com/36577992/217674172-2523f254-d8dd-4e22-8a41-ef0f2f0b343e.jpeg)
![69C1DC9B-9A32-4840-8DBE-4408623FF4FD](https://user-images.githubusercontent.com/36577992/217674175-8e299cd9-e5c6-4ad3-8770-be00d912fc0a.jpeg)
![6873DD69-3F8B-42FE-989E-A986C9419DFA](https://user-images.githubusercontent.com/36577992/217674179-d1b7f01b-d889-4fee-99fc-1e1d73fea214.jpeg)
![4DB76D5D-6807-47C2-8B12-BA6A700B043A](https://user-images.githubusercontent.com/36577992/217674183-5d42263b-755f-4c1d-8d76-b0d9e258b5ce.jpeg)
![Latest clip with notch](https://user-images.githubusercontent.com/36577992/217674184-d176e2b9-0669-4d7d-9f0b-7e17421e0c0f.jpeg)

**Comment by rbaron** [comment]
I'm always happy to hear about these success stories. Thanks for taking the time to write it down and for sharing the STL files.

> The b parasite shows 69% and the Mi Flora shows 32%

We have our very own very basic linear-ish model for moisture. I assume it's completely different from other sensors out there. I usually water my plants when they are below 30%.

> the issue with the sensors showing as "unavailable"

This is annoying now that more of us are using the BTHome integration. Hopefully we figure out a solution soon. A "workaround" is to lower the [PRST_SLEEP_DURATION_SEC](https://github.com/rbaron/b-parasite/blob/28db7224a68f38941b464adbf74de53c222e2367/code/nrf-connect/samples/ble/Kconfig#L5). Something like 180 (3 minutes) should give HA more opportunities to pick it up before it tags it as unavailable.

**Comment by BenRoe** [comment]
@dlavey any results with your UV solder mask ink on the edges test?
thanks for the clip stl. Will try it.
